net/libfab/libfab.c

```diff
@@ -1392,6 +1487,98 @@ static m0_bcount_t libfab_get_max_buf_desc_size(const struct m0_net_domain *dom)
 	return sizeof(uint64_t);
 }
 
+static struct m0_fab__tm  *libfab_buf_ma(struct m0_fab__buf  *buf)
+{
+	return buf->fb_nb->nb_tm->ntm_xprt_private;
+}
+
+static struct m0_fab__ep  *libfab_buf_ep(struct m0_net_end_point *net)
+{
+	return container_of(net, struct m0_fab__ep, fep_ep);;
+}
+
+/* Returns the IPV4 or IPV6 based on given ip string */
+static int libfab_ip_type(char *node,int *type)
+{
+	char	*cp;
+	int	 ret=0;
+	if( node != NULL ) {
+		cp = strchr(node, ':');
+		if( cp == NULL )
+			*type=AF_INET6;
+		else
+			*type=AF_INET;
+	}
+	else
+		ret=M0_ERR(-EFAULT);
+
+	return ret;
+}
+
+static int libfab_get_remote_addr(const char *addr,fi_addr_t *remote_rx_addr) 
+{
+	int	ret = 0;
+	int	ip_family;
+	char	node[INET6_ADDRSTRLEN];
+	char	port[8];
+
+	if(addr == NULL )
+		return M0_ERR(-EFAULT);
+
+	/* getting the addr of the remote */
+	ret = libfab_ep_addr_decode(addr, node, sizeof(node),
+				    port, sizeof(port));
+	if( ret == 0) {
+		ret = libfab_ip_type(node,&ip_family);
+		if( ret == 0) {
+			if( (inet_pton(ip_family, node, remote_rx_addr)) != 1)
```

Comment:
The inet_pton() will store a IPv4 addr (4 bytes) or IPv6 (16 bytes) to the destination buffer.
Is `remote_rx_addr ` ready to receive this address?
Is it large enough?
How does the `remote_rx_addr` know this is a IPv4 or IPv6?